### DISCUSSION TITLE
Suggest adding wasm32-unknown-unknown to cargo config

### DISCUSSION THREAD

**Discussion by pickfire** [DioxusLabs/dioxus#947] [General]
I noticed if I did not specify `wasm32-unknown-unknown` and I use `dioxus serve --hot-reload` with `rust-analyzer` running in the terminal, it needs a full build every time and the incremental build is very slow.

But if I add the wasm target to cargo config in `.cargo/config`, then it became very fast even on non-rsx changes (from ~48s to ~2.5s). Maybe because there are some issues with caching for rust-analyzer?

```toml
[build]
target = "wasm32-unknown-unknown"
```

Maybe we can suggest adding that to https://dioxuslabs.com/docs/0.3/guide/en/getting_started/hot_reload.html if web is used?

**Comment by ealmloff** [comment]
I just tested this on Windows on stable rust with the nightly and stable version of rust analyzer and builds with and without the target set were around 4-5s for a hello world example. That does sound like a issue with caching builds (a full rebuild for me takes ~30s). Maybe you on a different platform, version of rust, or version of rust analyzer?

**Reply by pickfire** [reply]
I tested this on arch linux, with stable rustc 1.68.2 (9eb3afe9e 2023-03-27), stable rust-analyzer 1 (236576227 2023-04-02) and 0.3.1 dioxus-cli.
